services/minespace-web/src/components/Forms/projects/majorMineApplication/MajorMineApplicationForm.tsx

```diff
@@ -0,0 +1,381 @@
+import React, { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { Field, change, getFormValues } from "redux-form";
+import "@ant-design/compatible/assets/index.css";
+import { Col, Row, Typography, Button } from "antd";
+import { required } from "@common/utils/Validate";
+import {
+  MAJOR_MINES_APPLICATION_DOCUMENT_TYPE,
+  MAJOR_MINES_APPLICATION_DOCUMENT_TYPE_CODE,
+} from "@mds/common/constants/strings";
+import { DOCUMENT, Feature, MODERN_EXCEL, SPATIAL, SPATIAL_DATA_STANDARDS_URL } from "@mds/common";
+import * as routes from "@/constants/routes";
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
+import { documentNameColumn, uploadDateColumn } from "@/components/common/DocumentColumns";
+import DocumentTable from "@/components/common/DocumentTable";
+import MajorMineApplicationFileUpload from "@/components/Forms/projects/majorMineApplication/MajorMineApplicationFileUpload";
+import ArchivedDocumentsSection from "@common/components/documents/ArchivedDocumentsSection";
+import { openModal } from "@mds/common/redux/actions/modalActions";
+import AddSpatialDocumentsModal from "@mds/common/components/documents/spatial/AddSpatialDocumentsModal";
+import SpatialDocumentTable from "@mds/common/components/documents/spatial/SpatialDocumentTable";
+import * as API from "@mds/common/constants/API";
+import { useFeatureFlag } from "@mds/common/providers/featureFlags/useFeatureFlag";
+import FormWrapper from "@mds/common/components/forms/FormWrapper";
+import { MajorMineApplicationDocument } from "@mds/common/models/documents/document";
+import { renderCategoryColumn } from "@mds/common/components/common/CoreTableCommonColumns";
+import * as Strings from "@mds/common/constants/strings";
+import { getMineDocuments } from "@mds/common/redux/selectors/mineSelectors";
+interface MajorMineApplicationFormProps {
+  project: any;
+  handleSubmit: () => void;
+  refreshData: () => void;
+}
+
+const MajorMineApplicationForm: React.FC<MajorMineApplicationFormProps> = ({
+  project,
+  handleSubmit,
+  refreshData,
+}) => {
+  const dispatch = useDispatch();
+
+  const formValues = useSelector(getFormValues(FORM.ADD_MINE_MAJOR_APPLICATION));
+
+  const { primary_documents = [], spatial_documents = [], supporting_documents = [] } =
+    formValues || {};
+
+  const mineDocuments = useSelector(getMineDocuments);
+  const [uploadedFiles, setUploadedFiles] = useState<any[]>([]);
+  const { isFeatureEnabled } = useFeatureFlag();
+
+  const acceptedFileTypesMap = {
+    ...SPATIAL,
+    ...DOCUMENT,
+    ...MODERN_EXCEL,
+  };
+
+  useEffect(() => {
+    if (!spatial_documents.length)
+      dispatch(change(FORM.ADD_MINE_MAJOR_APPLICATION, "spatial_documents", []));
+  }, [spatial_documents.length]);
```

Comment:
Just a recommendation to simplify things- in the FormWrapper you can put `initialValues={{ spatial_documents: [] }}` and then it'll initialize it how you want it to (assuming that's the purpose here).